config/v1/types_node.go

```diff
@@ -46,6 +46,14 @@ type NodeSpec struct {
 	// the status and corresponding reaction of the cluster
 	// +optional
 	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+
+	// MinimumKubeletVersion is the lowest version of a kubelet that can meaningfully join the cluster.
+	// Specifically, the apiserver will deny (most) authorization requests of kubelets that are older
+	// than the specified version.
+	// +kubebuilder:validation:Pattern=`^[0-9]*\.[0-9]*\.[0-9]*$`
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
```

Comment:
Since this field is not marked as immutable, what happens if I change it to value higher than existing kubelets?

Reply:
ack. I think we should probably clarify somewhere that this validation on admission is by the nature of a distributed system only a best effort. Inflight operations might still result in older kubelet bypassing it if the right lucky timing take place.

For hypershift yes the norm is signal via conditions. For improving consumer UX we might also want to explore introducing VAP and let the current minimal kubelet be reported via [vap parameter](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/#parameter-resources) that is inferred from e.g. the HC.status.

Reply:
> but set a condition on the hosted cluster object, rather than block on admission

Why can't we block on admission?